module/VuFind/src/VuFind/Cover/Loader.php

```diff
@@ -318,7 +319,7 @@ public function loadImage($settings = [])
         // Load settings from legacy function parameters if they are not passed
         // in as an array:
         $settings = is_array($settings)
-            ? array_merge($this->getDefaultSettings(), $settings)
+            ? $settings
```

Comment:
By factoring out the `getDefaultSettings()` merge here, we are now unable to guarantee that `$settings['title']`, `$settings['author']`, and `$settings['callnumber']` will be set below, which could trigger notices. It might be worth adding some `?? null` fallbacks to prevent that problem.

Reply:
`getDefaultSettings` is called from `storeSanitizedSettings` (which is called just few lines bellow), as you can't load image without calling storeSanitizedSettings, it should be OK I think...

Reply:
Ahh, yes, I see. I think I was confused because `storeSanitizedSettings` was changed to public. Is that actually necessary? If it still serves only as a support method for `loadImage`, it would probably be better to change it back to protected.

Reply:
`storeSanitizedSettings` is needed in cover router because the ability to get direct urls, so we need it to be public...

Reply:
Ahh, yes, I see. I think it would perhaps be less confusing if we could refactor the code slightly to avoid the need to call `storeSanitizedSettings` separately. That just feels to me like we're exposing too many class internals in public. How would you feel about adding:

<pre>
public function getIdentifiersForSettings($settings)
{
    $this->storeSanitizedSettings($settings);
    return $this->getIdentifiers();
}
</pre>

...instead of making `getIdentifiers` and `storeSanitizedSettings` public? I think that might make it easier to refactor the code in future if we want to make it more functional and less dependent on side effects.

Sorry if I'm being too picky here; this just feels like it might clarify things slightly.

Reply:
Thanks @demiankatz, this does make sense, I've refactored it that way.

Reply:
Great, thanks!